ipatests/test_integration/test_idm_api.py

```diff
@@ -0,0 +1,298 @@
+#
+# Copyright (C) 2018  FreeIPA Contributors see COPYING for license
+#
+from __future__ import absolute_import
+
+from ipaplatform.paths import paths
+from ipatests.test_integration.base import IntegrationTest
+from ipatests.pytest_ipa.integration import tasks
+from ipaplatform.tasks import tasks as platform_tasks
+from ipalib import api
+import pytest
+import textwrap
+
+
+class TestAPIScenario(IntegrationTest):
+    """
+    Tests for IDM API scenarios
+    """
+
+    topology = "line"
+
+    API_INIT = 'api.bootstrap_with_global_options(context="server")'
+    API_FIN = "api.finalize()"
+    API_CONNECT = "api.Backend.ldap2.connect()"
+
+    def test_idm_user_add(self):
+        user_code_script = textwrap.dedent(
+            """
+        from ipalib import api, errors
+        API_INIT
```

Comment:
if you are referencing existing constants, you need to interpolate these. So you need to change all these """....""" into f-strings:
```
f"""
  {API_INIT}
 {API_FIN}
...
"""
```

Reply:
To explain it with a bit of an example:
```
>>> FOO='text'
>>> f"""
... {FOO}
... """
'\ntext\n'
>>> f"""
... FOO
... """
'\nFOO\n'

```

Reply:
@abbra thank you for the review, will make the changes.